### PR TITLE
__user_groups: refactor, support FreeBSD

### PR DESCRIPTION
**Pull request by andres-erbsen**

### CODE REVIEW

File: cdist/conf/type/__user_groups/explorer/oldusermod [hunk 1]
```diff
@@ -22,7 +22,8 @@ os="$($__explorer/os)"
 
 if [ "$os" = "netbsd" ]; then
     echo netbsd
-    exit
+elif [ "$os" = "freebsd" ]; then
+    echo freebsd
+else
+    usermod --help | grep -q -- '-A group' && echo true || echo false
 fi
```
Comment (telmich) [inline cdist/conf/type/__user_groups/explorer/oldusermod]: Do we need the os = freebsd / netbsd here at all? If you catch on $os in gencode-remote, it looks like this value is not needed at all

Comment (darko-poljak) [inline cdist/conf/type/__user_groups/explorer/oldusermod]: @andres-erbsen Can you take a look?

**Comment by darko-poljak** [comment]
@andres-erbsen Can you take a look at the review and change/clean explorer accordingly?

**Comment by darko-poljak** [comment]
@telmich I made an explorer cleanup. Any (new) remarks or we should go with a merge?

File: cdist/conf/type/__user_groups/gencode-remote [hunk 2]
```diff
@@ -57,9 +42,25 @@ if [ -z "$changed_groups" ]; then
 fi
 
 for group in $changed_groups; do
-   if [ "$oldusermod" = "true" ]; then
-     echo "usermod $action \"$group\" \"$user\""
+   if [ "$os" = "netbsd" ]; then
```
Comment (telmich) [inline cdist/conf/type/__user_groups/gencode-remote]: So many cases -- can't we just have a case for netbsd and use the old mechanism for all other OS?

**Comment by darko-poljak** [comment]
@telmich FreeBSD does not have usermod at all.
And I believe there is a good reason why old usermode detection is added and new stuff used if usermod is not the old one.

**Pull request merged**